source/a_box/main.c

```diff
@@ -328,3 +332,19 @@ void PHAL_FaultHandler()
    asm("bkpt");
    HardFault_Handler();
 }
+
+void readCurrents() {
+    // storing ADC values to preserve them
+    volatile uint16_t adc_isense_1 = adc_readings.isense_ch1;
+    volatile uint16_t adc_isense_2 = adc_readings.isense_ch1;
```

Comment:
you have adc_readings.isense_ch1 twice

Reply:
that would explain why the readings are the same, fixed